src/components/Activity/ActivityPanel.js

```diff
@@ -1,20 +1,21 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
 import { Spring, animated } from 'react-spring'
+import { Button } from '@aragon/ui'
 import { ActivityContext } from '../../contexts/ActivityContext'
 import ActivityHub from './ActivityHub'
 import springs from '../../springs'
 
-const ActivityPanel = ({ open, onBlur, onClearAll }) => {
+const ActivityPanel = ({ getAppByProxyAddress, open, onBlur, onClearAll }) => {
   const frameRef = React.createRef()
   const { activities, clearActivity } = React.useContext(ActivityContext)
 
-  React.useEffect(() => {
+  useEffect(() => {
     frameRef.current[open ? 'focus' : 'blur']()
-  })
+  }, [frameRef.current, open])
 
-  const _onBlur = e => {
+  const handleBlur = e => {
     let _handler
     const currentTarget = e.currentTarget
     setTimeout(() => {
```

Comment:
@bpierre Do you know what the timeout is for?

Reply:
It is to prevent a click on the toggle button, while the panel is opened, to close it (because of the blur) and immediately reopening it (because of the click). But I had some weird behaviors that I suspect are caused by it, so I want to try a few alternatives to see if it fixes the issue.